### PR TITLE
Added JMX web server

### PR DESCRIPTION
**Pull request by HorGra**

### CODE REVIEW

File: distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void stop(){
+        try {
+            jmxWebServer.stop();
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (elbeyati) [inline distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java]: Don't ever do this again. Just throw the exception.

File: distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+public class LolStatsJmxWebServer {
+    private final int port;
+
+    private JmxWebServer jmxWebServer;
+
+    public LolStatsJmxWebServer(int port){
+        this.port = port;
+    }
+
+    public void initialise(){
```
Comment (elbeyati) [inline distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java]: ) {

File: distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.kh.lolstats;
+
+import com.j256.simplejmx.web.JmxWebServer;
+
+public class LolStatsJmxWebServer {
+    private final int port;
+
+    private JmxWebServer jmxWebServer;
+
+    public LolStatsJmxWebServer(int port){
```
Comment (elbeyati) [inline distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java]: ) {

File: distribution/src/main/java/com/kh/lolstats/LolStatsGatherer.java [hunk 4]
```diff
@@ -19,6 +19,7 @@
 
 public class LolStatsGatherer {
     private static final Logger logger = LoggerFactory.getLogger(LolStatsGatherer.class);
+    private static final int JMX_PORT = 8080;
```
Comment (elbeyati) [inline distribution/src/main/java/com/kh/lolstats/LolStatsGatherer.java]: This needs to be configurable, make it a program argument.

Comment (HorGra) [inline distribution/src/main/java/com/kh/lolstats/LolStatsGatherer.java]: done

File: distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java (hunk 3 above)
Comment (HorGra) [inline distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java]: x

File: distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java (hunk 2 above)
Comment (HorGra) [inline distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java]: x

File: distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    public LolStatsJmxWebServer(int port) {
+        this.port = port;
+    }
+
+    public void register(Object object) throws JMException {
+        jmxServer.register(object);
+    }
+
+    public void initialise(){
+        jmxWebServer = new JmxWebServer(port);
```
Comment (elbeyati) [inline distribution/src/main/java/com/kh/lolstats/LolStatsJmxWebServer.java]: Why is this not in the constructor?

File: distribution/src/main/java/com/kh/lolstats/LolStatsGatherer.java [hunk 6]
```diff
@@ -60,15 +56,20 @@ public void initialise() throws SQLException, FileNotFoundException, ClassNotFou
         matchTracker.initialise();
         persister.initialise();
 
-        ObjectName persisterObjectName = new ObjectName("com.kh.lolstats:type=Persister");
-        ObjectName apiKeyAccessorObjectName = new ObjectName("com.kh.lolstats.misc:type=ApiKeyAccessor");
-        mBeanServer.registerMBean(persister, persisterObjectName);
-        mBeanServer.registerMBean(apiKeyAccessor, apiKeyAccessorObjectName);
+        jmxWebServer.initialise();
```
Comment (elbeyati) [inline distribution/src/main/java/com/kh/lolstats/LolStatsGatherer.java]: Pull this out into a helper class, and generically add objects to be registered. Like a collection of objects.

File: persister/src/main/java/com/kh/lolstats/PersisterMBean.java [hunk 7]
```diff
@@ -1,7 +1,7 @@
 package com.kh.lolstats;
 
 public interface PersisterMBean {
```
Comment (elbeyati) [inline persister/src/main/java/com/kh/lolstats/PersisterMBean.java]: Is the MBean interface still necessary?

Comment (HorGra) [inline persister/src/main/java/com/kh/lolstats/PersisterMBean.java]: no. tested that it's not necessary.

**Pull request merged**